### PR TITLE
BUG: Fix compiled python module import on windows

### PR DESCRIPTION
**Pull request by jcfr**
This commit fixes a regression introduced in 34e48e8 (ENH: Upgrade
from python 3.6.7 to 3.9.10)

It ensures the "LibraryPaths" env. variables read by "slicer_dll_directories.add()"
is set when called from the "sitecustomize" module imported at the Slicer
application starts.

See https://docs.python.org/3/library/os.html#os.add_dll_directory

To ensure specifying `LibraryPaths` and `Paths` as `ADDITIONAL_PATH_ENVVARS`
does not overwrite the existing list of values, this commit also updates
the ctkAppLauncher CMake module provided by CTKAPPLAUNCHER (v0.1.29 to v0.1.30)

Note also that v0.1.30 of the pre-built `CTKAPPLAUNCHER` macOS package is
built targeting macOS 10.13 (instead of 10.6) and built against Qt 5.15.2 (instead
of 4.7.4). This is consistent with the Slicer macOS deployment target being
10.13.

List of `CTKAppLauncherLib` & `CTKAPPLAUNCHER` changes:

```
$ git shortlog 1367de4..7e87d4c --no-merges
Jean-Christophe Fillion-Robin (7):
      Begin post-v0.1.29 development [ci skip]
      BUG: Support specifying LibraryPaths and Paths as ADDITIONAL_PATH_ENVVARS
      BUG: Fix segfault in ctkAppLauncher buildEnvironment when build with Qt >= 5.14
      ENH: Transition macOS CI from travis to GitHub Actions
      ENH: Add support for automatically uploading macOS package as release asset
      BUG: Workaround publish_github_release upload issue using actions/upload-artifact
      CTKAppLauncher v0.1.30
```

### CODE REVIEW

**Comment by jcfr** [comment]
Working on addressing a side effect of this change causing the setting file to include an empty `LibraryPaths` section at the end of `SlicerLauncherSettings.ini` and `SlicerLauncherSettingsToInstall.ini`

To put things in perspective, here is `SlicerLauncherSettings.ini`:

```
[LibraryPaths]
size=0
```

 ... 

```
[General]
...

[Application]
...

[ExtraApplicationToLaunch]

...


[Environment]
additionalPathVariables=QT_PLUGIN_PATH,PYTHONPATH,LibraryPaths

[LibraryPaths]
1\path=<APPLAUNCHER_SETTINGS_DIR>/bin/.
...
19\path=/home/jcfr/Projects/Slicer-Release/JsonCpp-build/src/lib_json/.
size=19

[Paths]
1\path=<APPLAUNCHER_SETTINGS_DIR>/bin/.
2\path=/home/jcfr/Software/Qt5.10.0/5.10.0/gcc_64/bin
3\path=<APPLAUNCHER_SETTINGS_DIR>/lib/Slicer-4.13/cli-modules/.
4\path=/home/jcfr/Projects/Slicer-Release/python-install/bin
5\path=/home/jcfr/Projects/Slicer-Release/teem-build/bin/.
size=5

[EnvironmentVariables]
...
PYTHONHOME=/home/jcfr/Projects/Slicer-Release/python-install
PYTHONNOUSERSITE=1



[QT_PLUGIN_PATH]
...
3\path=/home/jcfr/Software/Qt5.10.0/5.10.0/gcc_64/plugins
size=3

[PYTHONPATH]
1\path=<APPLAUNCHER_SETTINGS_DIR>/bin/.
...
11\path=/home/jcfr/Projects/Slicer-Release/CTK-build/CTK-build/bin/.
size=11

[LibraryPaths]
size=0
````

**Comment by jcfr** [comment]
Once the fix pushed to the CTKAppLauncher is integrated, I will move forward with this PR.

See https://github.com/commontk/AppLauncher/pull/123

**Comment by jcfr** [comment]
Since macOS CI associated with the app launcher needs some work (`travis-ci.org` is not available anymore), here is the plan:
* [ ] Disable automatic start of nightly builds on `metroxplex`, `overload`, `computron` and `factory-south-macos`
* [ ] Add a patch step to `External_CTKAPPLAUNCHER.cmake` for downloading the updated CMake modules
* [ ] Manually trigger the nightly build by manually pulling the sources and disabling the update step

**Comment by jcfr** [comment]
> Add a patch step to `External_CTKAPPLAUNCHER.cmake` for downloading the updated CMake modules

I ended up fixing the macOS CI associated with `commontk/AppLauncher` project by transitioning from TravisCI to GitHub Action.

In the process, this also led me to:

* fix an issue happening when building the launcher against Qt >= 5.14. See https://github.com/commontk/AppLauncher/commit/ef13633cd89b59ee491a59dc0d18c2fa01ada220
* 
* publish a static build of Qt 5.15.2 for macOS. See https://github.com/jcfr/qt-static-build/releases/tag/qt-5.15.2-macosx10.13-static-x86_64

**Comment by lassoan** [comment]
@jcfr thanks a lot for all your work on this. There are many tickets in various repositories and it is hard to follow what is the current state of the master branch, what is expected to work, etc. Could you add a very short topic on Discourse in the Developer category answering these? Developers could follow the topic to get updates and notified when things get back to normal. Thank you!

**Comment by lassoan** [comment]
Awesome, thank you!
